setup.php

```diff
@@ -351,14 +355,26 @@
                 db_insert('domain', array('domain' => 'ALL', 'description' => '', 'transport' => '') ); // all other fields should default through the schema.
             }
 
-            $values = array(
-                'username'      => safepost('username'),
-                'password'      => safepost('password'),
-                'password2'     => safepost('password2'),
-                'superadmin'    => 1,
-                'domains'       => array(),
-                'active'        => 1,
-            );
+
+            if ( isset($_SERVER['REMOTE_USER']) ) {
+                $values = array(
+                    'username'      => strtolower($_SERVER['REMOTE_USER']),
+                    'password'      => safepost('password'),
+                    'password2'     => safepost('password2'),
+                    'superadmin'    => 1,
+                    'domains'       => array(),
+                    'active'        => 1,
+                );
+            } else {
+                $values = array(
```

Comment:
looks like duplication, I'd rather set all common values first and then add the differing value in an if clause